src/app/page.tsx

```diff
@@ -100,15 +101,29 @@ export default async function Main({ searchParams }: { searchParams: { token: st
         tokenPayload={tokenPayload}
         templates={templates}
       >
-        <DndWrapper>
-          <Header showCreateTaskButton={true} />
-          <FilterBar
-            updateViewModeSetting={async (payload: CreateViewSettingsDTO) => {
-              'use server'
-              await updateViewModeSettings(token, payload)
-            }}
-          />
-          <TaskBoard
+        <Header showCreateTaskButton={true} />
+        {tasks && tasks.length > 0 ? (
+          <DndWrapper>
+            <FilterBar
+              updateViewModeSetting={async (payload: CreateViewSettingsDTO) => {
+                'use server'
+                await updateViewModeSettings(token, payload)
+              }}
+            />
+            <TaskBoard
+              getSignedUrlUpload={async (fileName: string) => {
+                'use server'
+                return await getSignedUrlUpload(token, fileName)
+              }}
+              handleCreateMultipleAttachments={async (attachments: CreateAttachmentRequest[]) => {
+                'use server'
+                await createMultipleAttachments(token, attachments)
+              }}
+            />
+          </DndWrapper>
+        ) : (
+          <DashboardEmptyState
```

Comment:
I think we don't need to do this much just to solve the conditional rendering error.